enhancements/kubelet/split-filesystem.md

```diff
@@ -0,0 +1,315 @@
+to a lack of interest from customer requests.
+
+- Customers will not have the ability to change the location of the image cache.
+
+### Sketch of what happens when this feature is enabled
+
+- A user specifies in the container runtime config that they want to split the filesystem (use this feature)
+  - Feature gate is set for kubelet
+  - container storage uses image store (hard coded to /var/lib/images)
+  - /var/lib/images is relabeled to match the same selinux labels as /var/lib/container/storage.
+
+### Manual Enablement of Feature
+
+In the developer preview, a user can run the following steps to enable this feature.
+
+We will further refine the user APIs and interfaces for tech preview.
+
+#### Feature gate
+
+User can set the `KubeletSeparateDiskGC` feature gate in `CustomNoUpgrade.
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: FeatureGate
+metadata:
+  name: cluster
+spec:
+  featureSet: "CustomNoUpgrade"
+  customNoUpgrade:
+    enabled:
+      - KubeletSeparateDiskGC
+```
+
+
+#### Storage Configuration
+
+One could use a butane config as follows:
+
+```storage.bu
+variant: openshift
+version: 4.14.0
+metadata:
+  name: 40-storage-override
+  labels:
+    machineconfiguration.openshift.io/role: worker
+storage:
+  files:
+  - path: /etc/containers/storage.conf
+    mode: 0644
+    overwrite: true
+    contents:
+      inline: |
+        [storage]
+        
+        # Default Storage Driver
+        driver = "overlay"
+        
+        runroot = "/var/containers/storage"
+        graphroot = "/var/lib/containers/storage"
+        imageroot = "/var/lib/images"
+```
+
+And then run `butane storage.bu -o storage.yaml
+
+Applying storage.yaml will apply this machine config to your workers.
+
+#### Prequisites for Filesystem
+
+User must have a disk partition labeled as IMAGE_STORE.
+
+#### Script
+
+We will have a script called `split-filesystem.sh`.
+
+This bash script will mount the filesystem using `IMAGE_STORE` label.
+The script will check if the selinux labels are correct for `/var/lib/images`.
+If the labels are correct, we will create a file called `.relabel_complete`.
+If the labels are incorrect, folder will be relabled.
+
+This script will be injected as a systemd file and referenced in a unit file similarly to how we do this for the kubelet auto sizing feature.
+
+One can look at [kubelet-auto-sizing scruot](https://github.com/openshift/machine-config-operator/blob/master/templates/common/_base/files/kubelet-auto-sizing.yaml) for an example of the script and [kubelet-auto-sizing unit](https://github.com/openshift/machine-config-operator/blob/master/templates/common/_base/units/kubelet-auto-node-size.service.yaml) for an example of the unit file.
+
+We will run this script after the `crio.service`.
+
+#### Remove all old images
+
+Since the image cache has changed locations, all the old images left over should be removed.
+
+Simplest option is to remove the images on each node that this feature was enabled.
```

Comment:
should spell out what to do with the containers (which would be left over after the reboot), especially because a lot of those containers will require a running network pod to remove

Reply:
I have this sorta as an open question.

Do we manually delete all of /var/lib/container/storage on enablement of this feature or disable?

I think using a tool like `crictl` may actually not work if we change the container runtime. For podman changing these locations does orphan those containers/images.

I think a manual rm may be the best thing we can do.. Unless we delete before enablement of the feature.

Reply:
Why do we want the old images removed?